From d9d3ee5b9206c011aa51df58e6283913d800056b Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 19 Nov 2018 13:02:39 +0100
Subject: [PATCH] arm: dt: fsl-imx8qxp-tqma8qx-mba8qx: reorder nodes
 alphabetically

while at it, add iomux hog node (revision gpio)

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm/dts/fsl-imx8qxp-tqma8qx-mba8qx.dts | 59 +++++++++++++--------
 1 file changed, 37 insertions(+), 22 deletions(-)

diff --git a/arch/arm/dts/fsl-imx8qxp-tqma8qx-mba8qx.dts b/arch/arm/dts/fsl-imx8qxp-tqma8qx-mba8qx.dts
index 9d01afc901..71a45ebb09 100644
--- a/arch/arm/dts/fsl-imx8qxp-tqma8qx-mba8qx.dts
+++ b/arch/arm/dts/fsl-imx8qxp-tqma8qx-mba8qx.dts
@@ -91,12 +91,27 @@
 			>;
 		};
 
+		pinctrl_hog: mba8qxhoggrp {
+			fsl,pins = <
+				SC_P_MIPI_CSI0_I2C0_SCL_LSIO_GPIO3_IO05		0x06000048
+				SC_P_MIPI_CSI0_I2C0_SDA_LSIO_GPIO3_IO06		0x06000048
+				SC_P_MIPI_CSI0_GPIO0_01_LSIO_GPIO3_IO07		0x06000048
+			>;
+		};
+
 		pinctrl_pca9538: pca9538grp {
 			fsl,pins = <
 				SC_P_MIPI_CSI0_GPIO0_00_LSIO_GPIO3_IO08		0x06000020
 			>;
 		};
 
+		pinctrl_usbotg1: usbotg1grp {
+			fsl,pins = <
+				SC_P_USB_SS3_TC0_CONN_USB_OTG1_PWR	0x06000020
+				SC_P_USB_SS3_TC2_CONN_USB_OTG1_OC	0x06000020
+			>;
+		};
+
 		pinctrl_usdhc2_gpio: usdhc2gpiogrp {
 			fsl,pins = <
 				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x06000021
@@ -140,22 +155,9 @@
 				SC_P_USDHC1_VSELECT_CONN_USDHC1_VSELECT	0x06000020
 			>;
 		};
-
-		pinctrl_usbotg1: usbotg1grp {
-			fsl,pins = <
-				SC_P_USB_SS3_TC0_CONN_USB_OTG1_PWR	0x06000020
-				SC_P_USB_SS3_TC2_CONN_USB_OTG1_OC	0x06000020
-			>;
-		};
 	};
 };
 
-&lpuart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_lpuart1>;
-	status = "okay";
-};
-
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_fec1>;
@@ -175,6 +177,9 @@
 			reg = <0>;
 			ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_50_NS>;
 			ti,tx-internal-delay = <DP83867_RGMIIDCTL_2_50_NS>;
+			ti,dp83867-rxctrl-strap-quirk;
+			ti,led-function = <0xc100>; /* LED1: Link, LED2: activity */
+			ti,led-ctrl = <0x1001>;     /* active low, LED1/2 driven by phy */
 		};
 
 		ethphy3: ethernet-phy@3 {
@@ -182,6 +187,9 @@
 			reg = <3>;
 			ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_50_NS>;
 			ti,tx-internal-delay = <DP83867_RGMIIDCTL_2_50_NS>;
+			ti,dp83867-rxctrl-strap-quirk;
+			ti,led-function = <0xc100>; /* LED1: Link, LED2: activity */
+			ti,led-ctrl = <0x1001>;     /* active low, LED1/2 driven by phy */
 		};
 	};
 };
@@ -216,15 +224,10 @@
 	};
 };
 
-&usdhc2 {
-	pinctrl-names = "default", "state_100mhz", "state_200mhz";
-	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
-	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
-	bus-width = <4>;
-	cd-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
-	wp-gpios = <&gpio4 21 GPIO_ACTIVE_HIGH>;
-	vmmc-supply = <&reg_usdhc2_vmmc>;
+
+&lpuart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart1>;
 	status = "okay";
 };
 
@@ -243,3 +246,15 @@
 &usb2 {
 	status = "disabled";
 };
+
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
+	bus-width = <4>;
+	cd-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio4 21 GPIO_ACTIVE_HIGH>;
+	vmmc-supply = <&reg_usdhc2_vmmc>;
+	status = "okay";
+};
